### ISSUE TITLE
Slow Class Creation

### ISSUE THREAD

**Issue by agentgruer** [joeferraro/MavensMate#604]
It is currently taking me several minutes (6+ minutes) to create new class through Mavensmate, but saving class only takes a matter of seconds. I haven't changed any setting on my org or to Mavensmate, and my network is running fast. Is anyone else experiencing this, or is it just me?

**Comment by ralphcallaway** [comment]
i run into this frequently, but it's for both saves and class creation

in most cases after mavensmate is done compiling a class and retrieves that class from salesforce to update the editor with any changes from the compilation process, randomly the retrieve seems to take literally forever

i can't say for certain, but it appears that salesforce is just being really, really slow in returning the retrieve result, a couple times i was doing a deployment with ant and saw the same massive delay in returning retrieve results with ant that i was experience with mavensmate saves

in summary, yes i'm definitely experiencing it a lot, and it appears to a be a salesforce performance issue, not a mavensmate one

at least for me, if something is taking too long, i'll restart mavensmate to kill the operation, then refresh the file to see if it's saved (most of the time it was successful unless i had a compilation error)

**Comment by agentgruer** [comment]
Alright, thanks for the response. I didn't restart mavensmate, but the problem went away on its own. It must have been a SF issue.
